### PR TITLE
Fix misc gcc warnings

### PR DESCRIPTION
**Pull request by derobins**
* Remove dead code in hdf/src/mfan.c
* Add const to many constant string initializations
* Fix some integer type mismatch with casts
* Fix logic and parentheses in szip test
* Clarify ambiguous brackets

### CODE REVIEW

File: config/gnu-warnings/no-developer-general [hunk 1]
```diff
@@ -6,3 +6,4 @@
 # regarding the library settings string (H4libhdf4_settings). We'll turn
 # it off here but leave it on in the developer flags.
 -Wno-overlength-strings
+-Wno-format-nonliteral
```
Comment (derobins) [inline config/gnu-warnings/no-developer-general:9]: These are just noise. We have no plans to rework how the library works in a way that would make these go away.

File: hdf/test/tmgrcomp.c [hunk 2]
```diff
@@ -351,11 +348,11 @@ test_mgr_compress_c()
...
-    for (i = 0; i < 128; i++)
-        for (j = 0; j < 128; j++) {
-            image_buf[i][j][0] = (i + j) + 1;
-            image_buf[i][j][1] = (i + j) + 1;
-            image_buf[i][j][2] = (i + j) + 1;
+    for (uint8 i = 0; i < 128; i++)
+        for (uint8 j = 0; j < 128; j++) {
+            image_buf[i][j][0] = (uint8)(i + j + 1);
+            image_buf[i][j][1] = (uint8)(i + j + 1);
+            image_buf[i][j][2] = (uint8)(i + j + 1);
```
Comment (derobins) [inline hdf/test/tmgrcomp.c:355]: Fixed complaints about integer casts

File: hdf/test/tszip.c [hunk 3]
```diff
@@ -134,7 +134,7 @@ test_szip_RI8bit()
     /********************** End of variable declaration **********************/
 
     HCget_config_info(COMP_CODE_SZIP, &comp_config);
-    CHECK_VOID((comp_config & COMP_DECODER_ENABLED | COMP_ENCODER_ENABLED), 0,
+    CHECK_VOID(((comp_config & COMP_DECODER_ENABLED) && (comp_config & COMP_ENCODER_ENABLED)), 0,
```
Comment (derobins) [inline hdf/test/tszip.c:137]: I assume we want both flags set

**Review by bmribler — APPROVED** [review]

File: mfhdf/libsrc/mfsd.c [hunk 4]
```diff
@@ -3577,9 +3577,11 @@ SDsetup_szip_parms(int32 id, NC *handle, comp_info *c_info, int32 *cdims)
     }
 
     ndims = var->assoc->count;
-    for (i = 0; i < ndims; i++) {
+    for (int i = 0; i < ndims; i++) {
         dimindex = var->assoc->values[i];
         dim      = SDIget_dim(handle, dimindex);
+        if (dim == NULL)
+            HGOTO_ERROR(DFE_ARGS, FAIL);
```
Comment (derobins) [inline mfhdf/libsrc/mfsd.c:3584]: Fixes a potential NULL pointer dereference

File: mfhdf/ncdump/vardata.c [hunk 5]
```diff
@@ -565,6 +569,10 @@ vardata(struct ncvar *vp, long vdims[], int ncid, int varid, struct fspec *fsp)
                             printf(",%d", (int)(1 + cor[id]));
                         }
                         break;
+                    case LANG_NONE:
+                    default:
+                        /* Might want to make this an error */
+                        break;
```
Comment (derobins) [inline mfhdf/ncdump/vardata.c:575]: Fixes complaints about unhandled cases

File: mfhdf/test/tncunlim.c [hunk 6]
```diff
@@ -141,7 +141,7 @@ test_1dim_singlevar()
         long start[] = {0};
         long edges[1];
         edges[0] = dimsize;
-        memset(outdata, 0, DIM0);
+        memset(outdata, 0, DIM0 * sizeof(short));
```
Comment (derobins) [inline mfhdf/test/tncunlim.c:144]: memset was only cleaning half the bytes

**Pull request merged**